nutanix/structure.go

```diff
@@ -103,7 +104,113 @@ func flattenNicList(nics []*v3.VMNic) []map[string]interface{} {
 	return nicLists
 }
 
+func usesGuestCustomization(d *schema.ResourceData) bool {
+	keys := []string{
+		"guest_customization_cloud_init_user_data",
+		"guest_customization_cloud_init_meta_data",
+		"guest_customization_cloud_init_custom_key_values",
+		"guest_customization_is_overridable",
+		"guest_customization_sysprep",
+		"guest_customization_sysprep_custom_key_values"}
+	for _, k := range keys {
+		if _, ok := d.GetOk(k); ok {
+			return true
+		}
+	}
+	return false
+}
+
+func getDeviceIndexForDisk(disk *v3.VMDisk) (*int64, error) {
+	if disk.DeviceProperties == nil {
+		return nil, fmt.Errorf("deviceproperties was nil for disk")
+	}
+	if disk.DeviceProperties.DiskAddress == nil {
+		return nil, fmt.Errorf("disk address was nil for disk")
+	}
+	if disk.DeviceProperties.DiskAddress.DeviceIndex == nil {
+		return nil, fmt.Errorf("device index was nil for disk")
+	}
+	diskIndex := *disk.DeviceProperties.DiskAddress.DeviceIndex
+	return &diskIndex, nil
+}
+
+func flattenDiskListFilterCloudInit(d *schema.ResourceData, disks []*v3.VMDisk) ([]map[string]interface{}, error) {
+	//todo check if guestcust is passed -> if it is not passed, just continue without searching for cloud-init uuid
```

Comment:
can we do something like d.GetOK(guestCustom...)? to check if tha value is set?

Also can we parse the cloud init script to check if actually the cd room is creating by the scrip?